src/lib/features/segment/segment-controller.ts

```diff
@@ -372,9 +372,16 @@ export class SegmentsController extends Controller {
         res: Response,
     ): Promise<void> {
         const id = Number(req.params.id);
-        const strategies = await this.segmentService.getAllStrategies(id);
 
-        if (strategies.length > 0) {
+        let segmentIsInUse = false;
+        if (this.flagResolver.isEnabled('detectSegmentUsageInChangeRequests')) {
+            segmentIsInUse = await this.segmentService.isInUse(id);
+        } else {
+            const strategies = await this.segmentService.getAllStrategies(id);
+            segmentIsInUse = strategies.length > 0;
+        }
```

Comment:
Really not a big fan of reassigning variables like this, but because `if` is a statement and not an expression in JS, the alternative would be a pretty nasty ternary (see suggestion below). I think this version is cleaner for this use case. Plus, we'll remove it once the flag goes away.

```suggestion
        let segmentIsInUse = this.flagResolver.isEnabled(
            'detectSegmentUsageInChangeRequests',
        )
            ? await this.segmentService.isInUse(id)
            : (await this.segmentService.getAllStrategies(id)).length > 0;
```